### PR TITLE
docs: polish algorithm tutorial and update API docs

### PR DESCRIPTION
**Pull request by Gaiejj**
## Description

- Fix some mathematical formula.
- Fix grammar.
- Polish expression.
- Support API docs of statistics, plotter, adapters and simmer agents.
- Update usage video.

I have built the docs in [my fork](https://omnisafe-zjy.readthedocs.io/en/latest/)

image

Reviewers can see what I changed in this PR and review the docs directly in above link.

## Types of changes

What types of changes does your code introduce? Put an `x` in all the boxes that apply:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds core functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation (update in the documentation)

## Checklist

Go over all the following points, and put an `x` in all the boxes that apply.
If you are unsure about any of these, don't hesitate to ask. We are here to help!

- [x] I have read the [CONTRIBUTION](https://github.com/OmniSafeAI/omnisafe/blob/HEAD/CONTRIBUTING.md) guide. (**required**)
- [x] My change requires a change to the documentation.
- [ ] I have updated the tests accordingly. (*required for a bug fix or a new feature*)
- [x] I have updated the documentation accordingly.
- [x] I have reformatted the code using `make format`. (**required**)
- [x] I have checked the code using `make lint`. (**required**)
- [x] I have ensured `make test` pass. (**required**)

### CODE REVIEW

**Comment by zmsn-2077** [comment]
image

**Comment by zmsn-2077** [comment]
Poor expression: You can return to this section any time you are puzzled by some notations in the following chapters.

**Comment by zmsn-2077** [comment]
image
Lagrange multiplier

**Comment by zmsn-2077** [comment]
image
the number of vectorized environment

**Comment by zmsn-2077** [comment]
Lagrange Duality can refer to https://web.stanford.edu/~boyd/cvxbook/bv_cvxbook.pdf, page 216, can cite this book.

**Comment by zmsn-2077** [comment]
image
alignment

File: docs/source/baserl/ppo.rst [hunk 1]
```diff
@@ -20,14 +20,22 @@ PPO Theorem
...
-without stepping so far that we accidentally cause performance collapse?
+
+.. note::
+    How can we take as big as an improvement step on a policy update using the data
+    we already have,
+    without stepping so far that we accidentally cause performance collapse?
+
+
 Instead of solving this problem with a complex second-order method as TRPO do,
 PPO use a few other tricks to keep new policies close to old.
```
Comment (XuehaiPan) [inline docs/source/baserl/ppo.rst:35]:
```suggestion
PPO uses a few other tricks to keep new policies close to the old ones.
```

Should check English grammar first.

**Review by zmsn-2077 — APPROVED** [review]
LGTM.

**Review by muchvo — APPROVED** [review]
LGTM.

**Pull request merged**